fix: Regression in contentScope Feature After Merge

syuzanvs opened the pull request:
- Loops through all `.ck-content` elements.
- Concatenates their innerHTML with newline separation, ensuring the full context is captured correctly.

Closes #93

jjroelofs reviewed (changes requested):
## Root cause: Scope creep (literally)

I think there was a misunderstanding somewhere that is causing all these problems and more problems when we put this into production.

Earlier it was agreed that contentScope is defined like this, I'm copying what is in the README:

`CSS selector that extends context gathering to include content from other CKEditor 5 instances found within the first matching ancestor element`

The current version of contentScope **only includes contents inside the other CK5 instances** — Currently the code is doing a lot of things that don't make sense in that context: cloning the body tag (can come at huge memory costs in a CMS/Page Builder editing context!), removing style and script tags. We don't really expect style and script tags inside CKEditor instances right?
Nor do we expect any of those classnames inside CKEditor's model output.

The reason I support to only include CKEditor contents for now is that it is too difficult to predict what we need to clean. You list 3 CKEditor classes, but in the implementation in DXPR Builder, our page builder, we could easily add a few dozen more classes before we can call the HTML clean. And it is likely our client need to extend the list with their own classes. Therefore this is not a reliable solution.

It is also a solution we don't need based on the premise that we only enrich the contents with the contents of the other editors on within the selector. So that is the root cause we need to fix, and then we don't need all this cleaning logic and classes

![image](https://github.com/user-attachments/assets/151788bf-38ae-44b2-a144-4bfd07981e09)
. 

_I just want to emphasize, it is indeed good to have the AI understand the context better and include surrounding HTML, but it is not a priority now. In the future we will deal with it properly and come up with a solid plan for reliable clean context. We keep it out of scope for budgeting and practical reasons_

jjroelofs commented:
> ## Root cause: Scope creep (literally)
> I think there was a misunderstanding somewhere that is causing all these problems and more problems when we put this into production.
> 
> Earlier it was agreed that contentScope is defined like this, I'm copying what is in the README:
> 
> `CSS selector that extends context gathering to include content from other CKEditor 5 instances found within the first matching ancestor element`
> 
> The current version of contentScope **only includes contents inside the other CK5 instances** — Currently the code is doing a lot of things that don't make sense in that context: cloning the body tag (can come at huge memory costs in a CMS/Page Builder editing context!), removing style and script tags. We don't really expect style and script tags inside CKEditor instances right? Nor do we expect any of those classnames inside CKEditor's model output.
> 
> The reason I support to only include CKEditor contents for now is that it is too difficult to predict what we need to clean. You list 3 CKEditor classes, but in the implementation in DXPR Builder, our page builder, we could easily add a few dozen more classes before we can call the HTML clean. And it is likely our client need to extend the list with their own classes. Therefore this is not a reliable solution.
> 
> It is also a solution we don't need based on the premise that we only enrich the contents with the contents of the other editors on within the selector. So that is the root cause we need to fix, and then we don't need all this cleaning logic and classes
> 
> ![image](https://private-user-images.githubusercontent.com/904576/403710576-151788bf-38ae-44b2-a144-4bfd07981e09.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcwMDgzMTUsIm5iZiI6MTczNzAwODAxNSwicGF0aCI6Ii85MDQ1NzYvNDAzNzEwNTc2LTE1MTc4OGJmLTM4YWUtNDRiMi1hMTQ0LTRiZmQwNzk4MWUwOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNlQwNjEzMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYTc5NDUxOWE0ZDI0MGY0OTUwMTFmMWUwMWNjMDBlMTRjNTBhNDJlMWFjMTk2MzdlZjU0MzdhNzZlYTM4M2QwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.SQUgxbB0SbHsMoUiwMMQXrxRGwVGFVBJebNahBBtoVw) .
> 
> _I just want to emphasize, it is indeed good to have the AI understand the context better and include surrounding HTML, but it is not a priority now. In the future we will deal with it properly and come up with a solid plan for reliable clean context. We keep it out of scope for budgeting and practical reasons_

syuzanvs committed:
fix: Regression in contentScope Feature After Merge

- Loops through all .ck-content elements.
- Concatenates their innerHTML with newline separation, ensuring the full context is captured correctly.

Closes #93

jjroelofs reviewed (approved):
ok
